File: .Backend-old/SkyLink.NotificationService/Models/Notification.cs
using System.ComponentModel.DataAnnotations;

namespace SkyLink.NotificationService.Models
{
    public class Notification
    {
        public int Id { get; set; }

        [Required]
        public int UserId { get; set; }

        [Required]
        public string Message { get; set; } = string.Empty;

        [Required]
        public string Type { get; set; } = string.Empty; // BookingConfirmation, FlightDelay, FlightCancellation

        [Required]
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        [Required]
        public bool IsRead { get; set; } = false;

        [Required]
        public bool EmailSimulated { get; set; } = false;
    }
}

File: Backend/services/notification-service/src/models/Notification.ts
import { Schema, model, Document } from 'mongoose';

export interface INotification extends Document {
  userId: string;
  message: string;
  type: string;
  createdAt: Date;
  isRead: boolean;
  emailSimulated: boolean;
}

const NotificationSchema = new Schema<INotification>({
  userId: { type: String, required: true },
  message: { type: String, required: true },
  type: { type: String, required: true },
  createdAt: { type: Date, default: Date.now },
  isRead: { type: Boolean, default: false },
  emailSimulated: { type: Boolean, default: true }
});

export const Notification = model<INotification>('Notification', NotificationSchema);
